core/src/consensus.rs

```diff
@@ -321,13 +321,24 @@ impl Tower {
             if let Some(fork_stake) = stake_lockouts.get(&vote.slot) {
                 let lockout = fork_stake.stake as f64 / total_staked as f64;
                 trace!(
-                    "fork_stake {} {} {} {}",
+                    "fork_stake slot: {} lockout: {} fork_stake: {} total_stake: {}",
                     slot,
                     lockout,
                     fork_stake.stake,
                     total_staked
                 );
-                lockout > self.threshold_size
+                for (new_lockout, original_lockout) in
+                    lockouts.votes.iter().zip(self.lockouts.votes.iter())
```

Comment:
1) I think we should only care about lockouts doubling if they are beyond the nth vote
2) The `process_slot_vote_unchecked` could have popped off some votes. I think if it pops off any votes then there's no need to check threshold either, as I don't think you can pop off votes and also double lockout

Reply:
For 1. if popping off results in not doubling lockout, then why not just test for not doubling lockout which we should do anyway even if we don't pop.

For 2, yea I will bail early if we make it to a threshold slot.